src/main/java/services/RestoreFactory.java

```diff
@@ -363,6 +373,9 @@ public String getRestoreUrl(boolean overwriteCache) {
             builder.append("&since=").append(syncToken);
         }
         */
+
+        builder.append("&device_id=" + getSyncDeviceId());
```

Comment:
A bit surprised that there's no need to URLEncode this parameter, but not sure if tehre's magic somewhere in htis process.

Also: as a heads up, Spring has a quite nice Url utilities class you can use to generate url's and add parameters and stuff safely without having to worry about stuff http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/UriComponents.html